src/ImageSharp/Processing/Processors/Transforms/RotateInterpolationProcessor.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+using SixLabors.ImageSharp.Helpers;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors
+{
+    internal class RotateInterpolationProcessor<TPixel> : RotateProcessor<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        /// <inheritdoc/>
+        protected override void OnApply(ImageFrame<TPixel> source, Rectangle sourceRectangle, Configuration configuration)
+        {
+            if (this.OptimizedApply(source, configuration))
+            {
+                return;
+            }
+
+            int height = this.CanvasRectangle.Height;
+            int width = this.CanvasRectangle.Width;
+            Matrix3x2 matrix = this.GetCenteredMatrix(source, this.ProcessMatrix);
+            Rectangle sourceBounds = source.Bounds();
+
+            using (var targetPixels = new PixelAccessor<TPixel>(width, height))
+            {
+                Parallel.For(
+                    0,
+                    height,
+                    new ParallelOptions { MaxDegreeOfParallelism = 1 }, //configuration.ParallelOptions,
+                    y =>
+                    {
+                        Span<TPixel> targetRow = targetPixels.GetRowSpan(y);
+
+                        for (int x = 0; x < width; x++)
+                        {
+                            var transformedPoint = PointF.Rotate(new Point(x, y), matrix);
+                            var rounded = Point.Round(transformedPoint);
+                            if (sourceBounds.Contains(rounded.X, rounded.Y))
+                            {
+                                int ceilX = unchecked((int)MathF.Ceiling(transformedPoint.X));
+                                int ceilY = unchecked((int)MathF.Ceiling(transformedPoint.Y));
+                                int floorX = unchecked((int)MathF.Floor(transformedPoint.X));
+                                int floorY = unchecked((int)MathF.Floor(transformedPoint.Y));
+
+                                // Clamp sampling pixels to the source image edge
+                                ceilX = Math.Min(ceilX, sourceBounds.Width - 1);
+                                floorX = Math.Max(0, floorX);
+                                ceilY = Math.Min(ceilY, sourceBounds.Height - 1);
+                                floorY = Math.Max(0, floorY);
+
+                                // At the image edge, take the whole value of source
+                                float wx1, wx2, wy1, wy2;
```

Comment:
I'd really like to get this working for any passed in resampling algorithm rather than just bilinear.  Could we refactor this to use weighting algorithms? @antonfirsov What do you think?

Reply:
I can add an overload to the ImageExtensions.Rotate to include a IRotationResampler, and then factor out the resampling logic into a bi-linear interpolation implementation of that interface. This would be much like the Resize setup. I thought I should be able to use the resamplers that already exist there but I cannot get my head around how the resize processor works. Any pointers welcome.

Reply:
Resize essentially calculates a collection of sampled weights for rows and columns based on the given algorithm and the input output size, then a window (of length determined by the algorithm) within that collection is summed and the input vector is then multiplied by that sum. 

Our implementation is complicated to follow but it's a good sacrifice to make for the superior performance it delivers.

There's a good introduction to resampling here.

http://entropymine.com/imageworsener/resample/